src/psyclone/psyir/nodes/omp_directives.py

```diff
@@ -435,23 +444,60 @@ def end_string(self):
 class OMPParallelDirective(OMPRegionDirective):
     ''' Class representing an OpenMP Parallel directive. '''
 
+    _children_valid_format = ("Schedule, OMPDefaultClause, [OMPPrivateClause], "
+                              "[OMPReductionClause]*")
```

Comment:
Discussed, need to change this so that `OMPPrivateClause` is non-optional child and always child 2 for now.

@sergisiso will discuss the structure of directive's in PSyclone developers' meeting to determine children order for these (i.e. `Schedule` first or clauses first).

Reply:
I am still not sure what is the nicest solution here. Another option we talked about in to have a ClausesList which is mandatory and always proceed the directive body. And inside having the clauses/ or nothing if all are optional.

But since I am still unconvinced and you need this to continue with your other PR. I am happy with the current implementation and I will come back to this in a future issue/pr.

Reply:
I am still not sure what is the nicest solution here. Another option we talked about in to have a ClausesList which is mandatory and always proceed the directive body. And inside having the clauses/ or nothing if all are optional.